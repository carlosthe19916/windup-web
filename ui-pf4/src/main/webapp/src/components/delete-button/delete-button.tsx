import * as React from "react";
import { useState, useEffect } from "react";
import {
  Button,
  Modal,
  ButtonVariant,
  ModalVariant,
  TextInput,
} from "@patternfly/react-core";
import { TrashIcon } from "@patternfly/react-icons";

export interface DeleteButtonProps {
  objType: string;
  objID: string;
  messageMatch: string;
  isDisabled?: boolean;
  onDelete: () => void;
  onCancel?: () => void;
}

export const DeleteButton: React.FC<DeleteButtonProps> = ({
  objType,
  objID,
  messageMatch,
  isDisabled,
  onDelete,
  onCancel,
}) => {
  const [isModalOpen, setIsModalOpen] = useState(false);
  const [inputMatchValue, setInputMatchValue] = useState("");
  const [allowDeletion, setAllowDeletion] = useState(false);

  useEffect(() => {
    setAllowDeletion(
      messageMatch.toLowerCase() === inputMatchValue.toLowerCase()
    );
  }, [inputMatchValue, messageMatch]);

  const handleModalToggle = () => {
    setIsModalOpen((val: boolean) => !val);
  };

  const handleDelete = () => {
    handleModalToggle();
    onDelete();
  };

  const handleCancel = () => {
    handleModalToggle();

    if (onCancel) {
      onCancel();
    }
  };

  return (
    <React.Fragment>
      <Button
        variant={ButtonVariant.plain}
        aria-label="Delete"
        isDisabled={isDisabled}
        onClick={handleModalToggle}
      >
        <TrashIcon />
      </Button>
      <Modal
        variant={ModalVariant.small}
        title={`Confirm ${objType} deletion`}
        isOpen={isModalOpen}
        onClose={handleModalToggle}
        actions={[
          <Button
            key="delete"
            variant={ButtonVariant.danger}
            isDisabled={!allowDeletion}
            onClick={handleDelete}
          >
            Delete
          </Button>,
          <Button key="cancel" variant="link" onClick={handleCancel}>
            Cancel
          </Button>,
        ]}
      >
        <div className="pf-c-content">
          <p>
            Are you sure you want to delete the project '{objID}'? This will
            delete all resources associated with '{objID}' and cannot be undone.
            Make sure this is something you really want to do!
          </p>

          <p>Type '{messageMatch}' to confirm.</p>

          <TextInput
            value={inputMatchValue}
            type="text"
            onChange={(value) => setInputMatchValue(value)}
            aria-label="Text input match"
          />
        </div>
      </Modal>
    </React.Fragment>
  );
};
