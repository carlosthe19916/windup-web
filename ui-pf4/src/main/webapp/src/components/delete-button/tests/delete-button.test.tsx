import React from "react";
import { shallow } from "enzyme";
import { DeleteButton } from "../delete-button";

describe("DeleteButton", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <DeleteButton
        objType="Project"
        objID="master"
        messageMatch="delete"
        onDelete={jest.fn}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
