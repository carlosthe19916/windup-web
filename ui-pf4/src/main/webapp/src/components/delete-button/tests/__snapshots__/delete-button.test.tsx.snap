// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton Renders without crashing 1`] = `
<Fragment>
  <Button
    aria-label="Delete"
    onClick={[Function]}
    variant="plain"
  >
    <TrashIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
    />
  </Button>
  <Modal
    actions={
      Array [
        <Button
          isDisabled={true}
          onClick={[Function]}
          variant="danger"
        >
          Delete
        </Button>,
        <Button
          onClick={[Function]}
          variant="link"
        >
          Cancel
        </Button>,
      ]
    }
    appendTo={<body />}
    aria-describedby=""
    aria-label=""
    aria-labelledby=""
    className=""
    hasNoBodyWrapper={false}
    isOpen={false}
    onClose={[Function]}
    ouiaSafe={true}
    showClose={true}
    title="Confirm Project deletion"
    variant="small"
  >
    <div
      className="pf-c-content"
    >
      <p>
        Are you sure you want to delete the project '
        master
        '? This will delete all resources associated with '
        master
        ' and cannot be undone. Make sure this is something you really want to do!
      </p>
      <p>
        Type '
        delete
        ' to confirm.
      </p>
      <ForwardRef
        aria-label="Text input match"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </Modal>
</Fragment>
`;
